ogr/ogrsf_frmts/hana/ogrhanadatasource.cpp

```diff
@@ -0,0 +1,1722 @@
+    else
+        return StringFormat(
+            "DRIVER={%s};SERVERNODE=%s:%s;DATABASENAME=%s;UID=%s;"
+            "PWD={%s};CURRENTSCHEMA=\"%s\";CHAR_AS_UTF8=1;%s",
+            paramDriver, paramHost, paramPort, paramDatabase, paramUser,
+            paramPassword, paramSchema, JoinStrings(params, ";").c_str());
+}
+
+int GetSrid(odbc::ResultSet& resultSet)
+{
+    int srid = -1;
+    while (resultSet.next())
+    {
+        odbc::Int val = resultSet.getInt(1);
+        if (!val.isNull())
+        {
+            srid = *val;
+            break;
+        }
+    }
+    resultSet.close();
+    return srid;
+}
+
+int GetColumnSrid(
+    odbc::Connection& conn,
+    const CPLString& schemaName,
+    const CPLString& tableName,
+    const CPLString& columnName)
+{
+    CPLString sql =
+        "SELECT SRS_ID FROM SYS.ST_GEOMETRY_COLUMNS WHERE SCHEMA_NAME = ?"
+        " AND TABLE_NAME = ? AND COLUMN_NAME = ?";
+    odbc::PreparedStatementRef stmt = conn.prepareStatement(sql.c_str());
+    stmt->setString(1, odbc::String(schemaName));
+    stmt->setString(2, odbc::String(tableName));
+    if (columnName != nullptr)
+        stmt->setString(3, odbc::String(columnName));
+    return GetSrid(*stmt->executeQuery());
+}
+
+int GetColumnSrid(
+    odbc::Connection& conn, const CPLString& query, const CPLString& columnName)
+{
+    CPLString clmName = QuotedIdentifier(columnName);
+
+    CPLString sql = StringFormat(
+        "SELECT %s.ST_SRID() FROM (%s) WHERE %s IS NOT NULL", clmName.c_str(),
+        query.c_str(), clmName.c_str());
+
+    odbc::StatementRef stmt = conn.createStatement();
+    return GetSrid(*stmt->executeQuery(sql.c_str()));
+}
+
+int GetSridWithFilter(odbc::Connection& conn, const CPLString& whereCondition)
+{
+    CPLAssert(whereCondition != nullptr);
+
+    int ret = -1;
+
+    odbc::StatementRef stmt = conn.createStatement();
+    CPLString sql = StringFormat(
+        "SELECT SRS_ID FROM SYS.ST_SPATIAL_REFERENCE_SYSTEMS WHERE %s",
+        whereCondition.c_str());
+    odbc::ResultSetRef rsSrs = stmt->executeQuery(sql.c_str());
+    while (rsSrs->next())
+    {
+        odbc::Int val = rsSrs->getInt(1);
+        if (!val.isNull())
+        {
+            ret = *val;
+            break;
+        }
+    }
+    rsSrs->close();
+
+    return ret;
+}
+
+CPLString GetSrsWktById(odbc::Connection& conn, int srid)
+{
+    CPLString ret;
+    const char* sql = "SELECT DEFINITION FROM "
+                      "SYS.ST_SPATIAL_REFERENCE_SYSTEMS WHERE SRS_ID = ?";
+    odbc::PreparedStatementRef stmt = conn.prepareStatement(sql);
+    stmt->setInt(1, odbc::Int(srid));
+    odbc::ResultSetRef rs = stmt->executeQuery();
+    while (rs->next())
+    {
+        odbc::String wkt = rs->getString(1);
+        if (!wkt.isNull())
+        {
+            ret = *wkt;
+            if (!ret.empty())
+                break;
+        }
+    }
+    rs->close();
+
+    return ret;
+}
+
+OGRwkbGeometryType GetGeometryType(
```

Comment:
Yes, you are right. Unfortunately, there is no such metadata in HANA. What we can do is to use the LIMIT clause that of course has some drawbacks. What do you think?

Reply:
There's no magical solution. If for the use cases you want to address, the current solution is good enough, so be it. This should probably be documented though as a limitation in the driver doc page. And possibly add an open option where users could ask the driver to avoid this iteration and just expose wkbUnknown as the layer geometry type

Reply:
Thanks for your prompt reply.
As you suggested, I've added an open options called DETECT_GEOMETRY_TYPE  that controls the layer geometry type, see a6545b9cc2d37d669cca1c1e986c46c87158237f.